services/user.py

```diff
@@ -0,0 +1,46 @@
+from django.contrib.auth import get_user_model
+
+from db.models import User
+
+
+def create_user(
+    username: str,
+    password: str,
+    first_name: str = None,
+    last_name: str = None,
+    email: str = None,
+) -> User:
+    user_data = {"username": username, "password": password}
+    if email:
+        user_data["email"] = email
+    if first_name:
+        user_data["first_name"] = first_name
+    if last_name:
+        user_data["last_name"] = last_name
+    return get_user_model().objects.create_user(**user_data)
```

Comment:
you’re passing password as part of user_data. However, create_user() in Django expects password to be set separately, as it hashes it.